python/whylogs/experimental/core/metrics/callback_metric.py

```diff
@@ -0,0 +1,121 @@
+import logging
+from dataclasses import dataclass, field
+from itertools import chain
+from typing import Any, Callable, Dict, Optional
+
+from whylogs.core.datatypes import (
+    DataType,
+    Fractional,
+    Integral,
+    StandardTypeMapper,
+    String,
+    TypeMapper,
+)
+from whylogs.core.metrics import StandardMetric
+from whylogs.core.metrics.metrics import (
+    Metric,
+    MetricConfig,
+    OperationResult,
+    register_metric,
+)
+from whylogs.core.metrics.multimetric import MultiMetric, SubmetricSchema
+from whylogs.core.preprocessing import PreprocessedColumn
+
+logger = logging.getLogger(__name__)
+
+
+class DefaultSchema(SubmetricSchema):
+    def resolve(self, name: str, why_type: DataType, fi_disabled: bool = False) -> Dict[str, Metric]:
+        metrics: Dict[str, Metric] = {
+            "counts": StandardMetric.counts.zero(MetricConfig()),
+            "types": StandardMetric.types.zero(MetricConfig()),
+            "cardinality": StandardMetric.cardinality.zero(MetricConfig()),
+        }
+
+        if isinstance(why_type, Integral):
+            metrics["distribution"] = StandardMetric.distribution.zero(MetricConfig())
+            metrics["ints"] = StandardMetric.ints.zero(MetricConfig())
+            if not fi_disabled:
+                metrics["frequent_items"] = StandardMetric.frequent_items.zero(MetricConfig())
+        elif isinstance(why_type, Fractional):
+            metrics["distribution"] = StandardMetric.distribution.zero(MetricConfig())
+        elif isinstance(why_type, String) and not fi_disabled:
+            metrics["frequent_items"] = StandardMetric.frequent_items.zero(MetricConfig())
+
+        return metrics
+
+
+@dataclass(frozen=True)
+class CallbackMetricConfig(MetricConfig):
+    """Maps feature name to callable that computes it"""
+
+    callbacks: Dict[str, Callable[[Any], Any]] = field(default_factory=dict)
+    submetric_schema: SubmetricSchema = field(default_factory=DefaultSchema)
+    type_mapper: TypeMapper = field(default_factory=StandardTypeMapper)
+
+
+class CallbackMetric(MultiMetric):
+    def __init__(
+        self,
+        callbacks: Dict[str, Callable[[Any], Any]],
+        # discover these with resolver  submetrics: Dict[str, Dict[str, Metric]],  # feature name -> (namespace -> metric)
+        submetric_schema: Optional[SubmetricSchema] = None,
+        type_mapper: Optional[TypeMapper] = None,
+        fi_disabled: bool = False,
+    ):
+        super().__init__(dict())  # submetrics)
+        self._callbacks = callbacks
+        self._submetric_schema = submetric_schema or DefaultSchema()
+        self._type_mapper = type_mapper or StandardTypeMapper()
+        self._fi_disabled = fi_disabled
+
+    @property
+    def namespace(self) -> str:
+        return "callback"
```

Comment:
consider making the namespace configurable

Reply:
I think the 'udf' part is fine, and you get to name the udf function. We might strip it out when displaying in reports or in WhyLabs.